sigstore/sign.py

```diff
@@ -137,10 +137,12 @@ def _signing_cert(
         """Get or request a signing certificate from Fulcio."""
         # If it exists, verify if the current certificate is expired
         if self.__cached_signing_certificate:
-            if (
-                datetime.now(timezone.utc).timestamp()
-                > self.__cached_signing_certificate.cert.not_valid_after.timestamp()
-            ):
+            not_valid_after = self.__cached_signing_certificate.cert.not_valid_after
+            not_valid_after_timestamp = not_valid_after.replace(
+                tzinfo=timezone.utc
+            ).timestamp()
+
+            if datetime.now(timezone.utc).timestamp() > not_valid_after_timestamp:
```

Comment:
Hmm, I think we can simplify this even further by just comparing the `datetime` objects directly: `cryptography` guarantees that `Certificate` times are always UTC, so this should work:

```python
if datetime.now(timezone.utc) > self.__cached_signing_certificate.cert.not_valid_after:
    ...
```

That way we don't have to do any timestamp conversions at all 🙂 

Could you confirm that this works for you, @perone?

Reply:
(NB: This might fail because of an aware vs. non-aware comparison, in which case we can use `datetime.utcnow()` instead of `datetime.now(timezone.utc)`).

Reply:
Thanks for reviewing @woodruffw. I would avoid the `utcnow()` [[1]](https://www.aaronoellis.com/articles/python-datetime-utcnow-considered-harmful) [[2]](https://blog.ganssle.io/articles/2019/11/utcnow.html). I will test it to check how it behaves.

Reply:
Indeed it triggers an exception:
```
TypeError: can't compare offset-naive and offset-aware datetimes
```

Another alternative is to go with the option of having both tz-aware datetimes but without timestamp conversion:

```python
not_valid_after = self.__cached_signing_certificate.cert.not_valid_after
not_valid_after_tzutc = not_valid_after.replace(
    tzinfo=timezone.utc
)
if datetime.now(timezone.utc) > not_valid_after_tzutc:
    raise ExpiredCertificate
```

What do you think ?

Reply:
That works for me -- frankly I'd also be okay with `utcnow` despite the footguns, since equality isn't defined between the two kinds of datetimes (the `TypeError` you noticed) and it's what `cryptography` uses internally (_and_ this object never gets exposed anywhere).

That being said, I'm fine with either, so whichever you prefer to do!

Reply:
(I take back what I said about `utcnow` being acceptable -- Python is marking it deprecated starting with 3.12, so we shouldn't use it at all. My bad!)